enhancements/testing/openshift-tests-extension.md

```diff
@@ -0,0 +1,854 @@
+
+// If we don't support their testing framework, they can wrap tests
+// with lower level API. Here, they give us an interface implementation
+// exposing .RunTests(testName ...string) []TestResult whose only contract is to run those
+// tests in parallel and return results in our required format.
+var customExtensionSpecs extensions.ExtensionTestSpecSet
+customExtensionSpecs, err = extensions.BuildExtensionTestSpec(myTestRunner, "test name 1", "test name 2", ...names this runner can run...))
+
+// Once individually wrapped in ExtensionTestSpec and grouped into ExtensionTestSpecSet
+// handling the ginkgo vs custom tests becomes identical. ExtensionTestSpec (ETS) is where you
+// can store all of the metadata that the enhancement calls for on a test-by-test basis.
+// As you note in your feedback, we can't possibily extract this from ginkgo tags, but 
+// we can, over time, expect more metadata to be attached by the author.
+
+// label all ETS in this set as slow.. Common ones, custom ones, it doesn't matter.
+// BuildExtensionTestSpecsFromGinkgoSuite may pre-populate tags using the regex [.*]
+// on tests names, while the lower-level API may not make this assumption.
+customExtensionSpecs.Label(commonlabels.SLOW) 
+
+// similar story with tags.
+customExtensionSpecs.Tag({"mytag": "myvalue"})  
+
+// Here, OpenShift convention was used to determine which tests to run on AWS
+// As a contrived example, let's say we want to run these same tests on GovCloud
+awsTests := ginkgoExtensionSpecs.Select(extensions.PlatformSelector(platforms.AWS))
+awsTests.AddPlatform(platforms.GovCloud)
+
+// And so on with expressive grouping & metadata management. awsTests
+// references the same objects as ginkgoExtensionSpecs, so changes in
+// one are reflected immediately in both sets.
+awsTests.Select(extensions.LabelSelector(platform.SLOW)).SetResource(resource.CPU, "5000m")
+
+// Since this set of tests doesn't use the OpenShift naming conventions, they
+// must constrain environmental parameters themselves.
+customExtensionSpecs.Select(extensions.NameRegExSelector(".*cld/aws.*")).SetPlatforms(platforms.AWS)
+
+// Since everything is an ExtensionTestSpec, you can mix sets.
+allSpecsSet := customExtensionSpecs
+allSpecsSet.Extend(ginkgoExtensionSpecs)
+
+// The author can iterate through all tests and, for example,
+// set OriginalName for each based on a configuration file
+// or dictionary maintained in their repository.
+allSpecsSet.Walk(func(*ExtensionTestSpec))
+
+// Finally, some or all of the specs are added for the component. 
+componentExtension.AddTestSpecSets(allSpecsSet)
+
+customSuite := componentExtension.DefineSuite("my-custom-suite")
+customSuite.AddQualifier("(test.tags.mytag==\"myvalue\"")  // CEL expression which selects appropriate tests regardless of the extension which defines the test
+customSuite.AddParent("openshift/conformance") // Also run this suite when openshift/conformance runs

```

Comment:
For now users just create a Suite struct, that seems more aligned with go not using excessive setters/getters